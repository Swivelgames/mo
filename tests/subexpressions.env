x="repo"
i=2
repo=( "resque" "hub" "rip" )
quote() {
    echo "'${MO_FUNCTION_ARGS[@]}'"
}
double_quote() {
    echo "\"${MO_FUNCTION_ARGS[@]}\""
}
inter_examp() {
    args=($MO_FUNCTION_ARGS)
    first_t=${args[0]}
    second_t=${args[1]}
    first_v=${args[2]}
    second_v=${args[3]}

    [[ "$first_t" == "true" ]] && echo -n "$first_v "
    [[ "$second_t" == "true" ]] && echo -n "$second_v "
}
exists="hello"
#doesnt="foo"
