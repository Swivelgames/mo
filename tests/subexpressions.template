Function: {{quote (x)}}
Specific Element: {{repo.(i)}}
Loop:
{{#repo}}
  <b>{{double_quote (quote (.))}} {{.}}</b>
{{/repo}}

{{inter_examp (#exists) (^doesnt) (exists) world}}
X: {{x}}
