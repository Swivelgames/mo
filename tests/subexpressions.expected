Function: 'repo'
Specific Element: rip
Loop:
  <b>"'resque'" resque</b>
  <b>"'hub'" hub</b>
  <b>"'rip'" rip</b>

hello world 
X: repo
